lib/bios/menus.robot

```diff
@@ -412,6 +413,48 @@ Reset To Defaults Tianocore
     Read From Terminal Until    ignore.
     Write Bare Into Terminal    y
 
+    # For platforms using PiKVM there is a need to turn on Serial Redirection otherwise most of tests might break.
+    IF    '${DUT_CONNECTION_METHOD}' == 'pikvm'
+        ${menu}=    Read From Terminal Until    Save screenshot
+        IF    '''Continue''' not in '''${menu}'''
+            ${menu}=    Go To Dasharo Main Screen
+        END
+        ${main_menu}=    Parse Menu Snapshot Into Construction    ${menu}    3    1
+        ${dasharo_menu}=    Enter Dasharo System Features    ${main_menu}
+        ${serial_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Serial Port Configuration
+        ${serial_state}=    Get Option State    ${serial_menu}    Enable COM0 Serial
+        IF    ${serial_state} != ${TRUE}
+            Set Option State    ${serial_menu}    Enable COM0 Serial    ${TRUE}
+        END
+        Press Key N Times    3    ${ESC}
+    END
+
+Go To Dasharo Main Screen
```

Comment:
You are aiming very high choosing to develop such a keyword.

It will be very tricky to get it right, depending on how it's been called.

In the first iteration this keyword relies on whatever was left in the telnet buffer by the previous keyword. The fact that you kewyord runs at all is simply because you chose to read until `save screenshot`, while in other menu handling keywords we read until `Esc=Exit` (`Save screenshot` has been added later at some point).

Reply:
https://github.com/Dasharo/open-source-firmware-validation/issues/540